stdlib/join/sort_merge_join.go

```diff
@@ -33,6 +33,21 @@ func (p *SortMergeJoinProcedureSpec) Copy() plan.ProcedureSpec {
 	}
 }
 
+func (p *SortMergeJoinProcedureSpec) RequiredAttributes() []plan.PhysicalAttributes {
+	return []plan.PhysicalAttributes{
+		{
+			plan.CollationKey: &plan.CollationAttr{
+				Columns: getJoinKeyCols(p.On, true),
+			},
+		},
+		{
+			plan.CollationKey: &plan.CollationAttr{
+				Columns: getJoinKeyCols(p.On, false),
+			},
+		},
+	}
+}
+
```

Comment:
Is the list of required attributes an `or` relationship? As in "predecessors must have one of these attributes, but not all"? Or is each required attribute somehow keyed to a specific parent node?

Reply:
Yes, each attribute (really a map of attributes) for each predecessor input to the join node. So this is requiring that the LHS of the join be sorted by the LHS join columns, and the RHS of the join must be sorted by the RHS join columns.

This assumes that the predecessors are ordered with `node.Predecessors()[0]` always being the left and `node.Predecessors()[1]` always being the right. That should hold true, right?

Reply:
Ah, I see. Didn't realize each item was a map. That makes much more sense now.

And yes, that assumption should always hold true.